src/Middleware/SubstituteBindings.php

```diff
@@ -68,9 +67,15 @@ protected function substituteImplicitBindings(Route $route)
             $class = $parameter->getClass()->getName();
 
             if ($repository = $this->registry->getRepository($class)) {
-                if ($parameter->getClass()->implementsInterface(UrlRoutable::class)) {
+                if ($parameter->getClass()->implementsInterface(\LaravelDoctrine\ORM\Contracts\UrlRoutable::class)) {
                     $name = call_user_func([$class, 'getRouteKeyName']);
 
+                    $entity = $repository->findOneBy([
+                        $name => $id
+                    ]);
+                } elseif ($parameter->getClass()->implementsInterface(\Illuminate\Contracts\Routing\UrlRoutable::class)) {
+                    $name = call_user_func([new $class, 'getRouteKeyName']);
```

Comment:
This will probably not work for everyone.

Reply:
Yes, the same problem arises when implemeting `\Illuminate\Contracts\Routing\UrlRoutable` on an entity with a constructor requiring parameters, i'll have to research how Doctrine constructs entities without calling their constructor.

Reply:
https://github.com/doctrine/orm/blob/2.7/lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php#L913

Though I believe it would be better to experiment with the alternative pointed out here: https://github.com/laravel-doctrine/orm/issues/451#issuecomment-665449403 
Then implemeneting `Illuminate\Contracts\Routing\UrlRoutable` would not be needed.